### DISCUSSION TITLE
Raspberry Pi 5 Support?

### DISCUSSION THREAD

**Discussion by leolilley** [josephernest/SamplerBox#64] [Q&A] [answered]
Apologies if this is not the right place for this. Is/will there be support for the Raspberry Pi 5?
Recently bought one and was hoping to run this on it but as far as I've tried I can't get it to work.
Any help would be much appreciated thanks.

**Comment by dosas** [comment]
The problem is not the raspberry board version but the python version running on the latest rsp image. This version is >= 3.9 which is currently not supported by ` rtmidi-python`

see also https://github.com/josephernest/SamplerBox/issues/58

**Comment by josephernest** [comment, marked as answer]
Thanks @leolilley! 

I have updated a few things since yesterday, and now it should work fine with Python 3.9+.

Can you retry all the installation instructions from https://github.com/josephernest/SamplerBox/blob/master/README.md and post if it works for you? Thanks in advance!

**Reply by leolilley** [reply]
Hey thanks for the quick reply. Yeah that has worked for me! Had to do a bit of wrangling to get the python env setup but I'm getting sound from the pi using my midi device. Although I do get a `MidiInAlsa::alsaMidiHandler: unknown MIDI input error!` error it seems to still work anyway. Cheers

**Reply by josephernest** [reply]
Thanks a lot for the report @leolilley! A few questions, so I'll be able to improve the documentation:

1) Did you do the last tests you mentioned with a RPi5? If so, it means that RPi5 is supported, great news :)

2) > Had to do a bit of wrangling to get the python env setup

    Can you add some detail? Maybe some additional `apt install`, do you remember which packages exactly? I'll update the doc if needed :)

3) Did you also test the last ZIP image from here with your Rpi5: https://www.samplerbox.org/makeitsoftware ? Did this one fail ?

Thanks!

**Reply by leolilley** [reply]
No worries, happy to help out.

1. Yes, I did it with a RPi5 just running the sampler Box from a terminal. The error message is shown in the terminal I'm running it from.
2. For the python setup I had to install the pip dependencies with `--break-system-packages`. There's probably a way to do it without this as it's not the recommended way to install pip packages, you'd just need to set up the environment to have the correct permissions to run everything.
3. I have tested that iso image. It doesn't work for the RPi5 as it complains that the iso image used is outdated.

**Comment by Alexus-3000** [comment]
PLEASE!!!
Lay out a working compilation scheme!!!!!
Does not work on any system - desktop, RPi - no difference

**Reply by josephernest** [reply]
Hmm, are you joking or what?

> Lay out a working compilation scheme!!!!!

Should I also give you 1000$ maybe?

I can't believe it... People giving orders, in an agressive tone, to people who publish open source code for free in their spare time. Very funny...

**Reply by zachroeder** [reply]
I suggest you follow some GitHub projects and see how others ask for help before commenting again. People here are so generous with their time and expert knowledge, which they provide for free. 

Asking for help is okay, but consider how to ask for help more effectively and respectfully.

**Reply by RobCranfill** [reply]
Say no more:
![image](https://github.com/user-attachments/assets/40115d1a-d4bd-406f-b54f-9a3c28dabec1)

**Comment by Alexus-3000** [comment]
Is the word "please!" an aggressive tone for you?
The world has gone crazy...
